UnityProject/Assets/Scripts/Health/RespiratoryHealth/RespiratorySystem.cs

```diff
@@ -268,6 +370,146 @@ private void HandleBreathingPlasma(float plasmaAmount)
+				: "its";
+			Chat.AddActionMsgToChat(
+				gameObject,
+				CO2LowYouMessages.PickRandom(),
+				string.Format(
+					CO2LowOthersMessages.PickRandom(),
+					gameObject.ExpensiveName(),
+					string.Format(plasmaLowOthersMessages.PickRandom(), gameObject.ExpensiveName(), theirPronoun))
+			);
+		}
+		// enough carbon dioxide to start suffocating you!
+		else if (carbonPressure > CARBON_DIOXIDE_SAFE_MAX)
+		{
+			if (CarbonSuffocationTimer >= 0)
+				CarbonSuffocationTimer++;
+			if (CarbonSuffocationTimer >= 3 && CarbonSuffocationTimer < 7)
+			{
+				IsSuffocating = true;
+				bloodSystem.OxygenDamage += 3;
+
+				if (DMMath.Prob(90))
+				{
+					return true;
+				}
+
+				// 10% chances of message
+				var theirPronoun = gameObject.Player() != null
+					? gameObject.Player().Script.characterSettings.TheirPronoun()
+					: "its";
+				Chat.AddActionMsgToChat(
+					gameObject,
+					CO2MedYouMessages.PickRandom(),
+					string.Format(CO2MedOthersMessages.PickRandom(), gameObject.ExpensiveName(), theirPronoun)
+				);
+			}
+			else if (CarbonSuffocationTimer >= 7)
+			{
+				IsSuffocating = true;
+				bloodSystem.OxygenDamage += 8;
+
+				if (DMMath.Prob(90))
+				{
+					return true;
+				}
+
+				// 10% chances of message
+				var theirPronoun = gameObject.Player() != null
+					? gameObject.Player().Script.characterSettings.TheirPronoun()
+					: "its";
+				Chat.AddActionMsgToChat(
+					gameObject,
+					CO2HighYouMessages.PickRandom(),
+					string.Format(CO2HighOthersMessages.PickRandom(), gameObject.ExpensiveName(), theirPronoun)
+				);
+				return true;
+			}
+		}
+
+		if (carbonPressure < CARBON_DIOXIDE_WARNING_LEVEL)
+		{
+			CarbonSuffocationTimer = 0;
+			return (carbonPressure > 0);
+		}
+		return false;
+	}
+	private bool HandleWearingGasMask(GasMix gasMix)
+	{
+		bool filtered = false;
+		// if there is too much CO2 in the air
+		if (gasMix.GetMoles(Gas.CarbonDioxide) >= 30)
+		{
+				GasMix gasMix2 = gasMix;
+				gasMix2.RemoveGas(Gas.CarbonDioxide, 30);
+				HandleBreathingCarbonDioxide(gasMix2);
+			filtered = true;
+		}
+		// if there is too much plasma in the air
+		if (gasMix.GetMoles(Gas.Plasma) >= 25)
+		{
+				GasMix gasMix2 = gasMix;
+				gasMix2.RemoveGas(Gas.Plasma, 25);
+				float plasmaBreathedWithMask = HandleBreathingPlasma(gasMix2);
+				if (plasmaBreathedWithMask > 0)
+				{
+					gasMix.RemoveGas(Gas.Plasma, plasmaBreathedWithMask);
+					registerTile.Matrix.MetaDataLayer.UpdateSystemsAt(registerTile.LocalPositionClient, SystemType.AtmosSystem);
+			}
+			filtered = true;
+		}
+		
+
+		if((gasMix.GetMoles(Gas.Plasma) < PLASMA_WARNING_LEVEL && gasMix.GetMoles(Gas.Plasma) > 0) || (gasMix.GetMoles(Gas.CarbonDioxide) < CARBON_DIOXIDE_WARNING_LEVEL && gasMix.GetMoles(Gas.CarbonDioxide) > 0))
+		{
+			return true;
+		}
+
+		if(gasMix.GetMoles(Gas.Plasma) == 0 && gasMix.GetMoles(Gas.CarbonDioxide) == 0)
+		{
+			return false;
+		}
+
+		if (DMMath.Prob(90))
```

Comment:
this bit is weird, why are you returning !filtered?

Reply:
you make a good point, seems i mistakenly told it to return the wrong value. i will remedy this really quick

Reply:
thanks for catching that